app/src/main/java/ch/sheremet/dasfoo/rover/android/client/MainActivity.java

```diff
@@ -134,5 +140,33 @@ protected void onPostExecute(final String result) {
             mResultText.setText(result);
             enableButtons(Boolean.TRUE);
         }
+
+        @Override
+        public void onProviderInstalled() {
+            Toast toast = Toast.makeText(MainActivity.this, PROVIDER_INSTALLED, Toast.LENGTH_SHORT);
+            toast.show();
+        }
+
+        @Override
+        public void onProviderInstallFailed(final int errorCode, final Intent intent) {
+            GoogleApiAvailability googleAPI = GoogleApiAvailability.getInstance();
+            if (googleAPI.isUserResolvableError(errorCode)) {
+                googleAPI.getErrorDialog(MainActivity.this, errorCode,
+                        PLAY_SERVICES_RESOLUTION_REQUEST).show();
+                return;
+            }
+            AlertDialog.Builder dialog = new AlertDialog.Builder(MainActivity.this);
+            dialog.setMessage(PROVIDER_NOT_INSTALLED);
```

Comment:
Is it possible to somehow convert that errorCode to a message and show it to the user?